python/cucim/src/cucim/skimage/metrics/tests/test_structural_similarity.py

```diff
@@ -115,30 +124,51 @@ def test_structural_similarity_multichannel():
     # replicate across three channels.  should get identical value
     Xc = cp.tile(X[..., cp.newaxis], (1, 1, 3))
     Yc = cp.tile(Y[..., cp.newaxis], (1, 1, 3))
-    S2 = structural_similarity(Xc, Yc, multichannel=True, win_size=3)
+
+    # move channels from last position to specified channel_axis
+    Xc, Yc = (cp.moveaxis(_arr, -1, channel_axis) for _arr in (Xc, Yc))
+
+    S2 = structural_similarity(Xc, Yc, channel_axis=channel_axis, win_size=3)
     assert_almost_equal(S1, S2)
 
     # full case should return an image as well
-    m, S3 = structural_similarity(Xc, Yc, multichannel=True, full=True)
+    m, S3 = structural_similarity(Xc, Yc, channel_axis=channel_axis, full=True)
     assert_equal(S3.shape, Xc.shape)
 
     # gradient case
-    m, grad = structural_similarity(Xc, Yc, multichannel=True, gradient=True)
+    m, grad = structural_similarity(Xc, Yc, channel_axis=channel_axis,
+                                    gradient=True)
     assert_equal(grad.shape, Xc.shape)
 
     # full and gradient case
-    m, grad, S3 = structural_similarity(
-        Xc, Yc, multichannel=True, full=True, gradient=True
-    )
+    m, grad, S3 = structural_similarity(Xc, Yc,
+                                        channel_axis=channel_axis,
+                                        full=True,
+                                        gradient=True)
     assert_equal(grad.shape, Xc.shape)
     assert_equal(S3.shape, Xc.shape)
 
     # fail if win_size exceeds any non-channel dimension
     with pytest.raises(ValueError):
-        structural_similarity(Xc, Yc, win_size=7, multichannel=False)
+        structural_similarity(Xc, Yc, win_size=7, channel_axis=None)
+
+
+def test_structural_similarity_multichannel_deprecated():
+    N = 100
+    X = (cp.random.rand(N, N) * 255).astype(np.uint8)
+    Y = (cp.random.rand(N, N) * 255).astype(np.uint8)
+
+    S1 = structural_similarity(X, Y, win_size=3)
+
+    # replicate across three channels.  should get identical value
+    Xc = cp.tile(X[..., cp.newaxis], (1, 1, 3))
+    Yc = cp.tile(Y[..., cp.newaxis], (1, 1, 3))
+    with expected_warnings(["`multichannel` is a deprecated argument"]):
+        S2 = structural_similarity(Xc, Yc, multichannel=True, win_size=3)
+    assert_almost_equal(S1, S2)
 
 
-@pytest.mark.parametrize('dtype', [cp.uint8, cp.float32, cp.float64])
+@pytest.mark.parametrize('dtype', [np.uint8, np.float32, np.float64])
```

Comment:
I wonder why some code is changed to `cp.uint8` and some other code is changed to `np.uint8`.

Reply:
updated in 8438d64